app/models/patient.rb

```diff
@@ -70,6 +70,13 @@ def vaccination_records_for_session(session)
     patient_sessions.find_by_session_id(session.id).vaccination_records
   end
 
+  def year_group
+    first_school_year = date_of_birth.year + 5
+    first_school_year += 1 if date_of_birth.month >= 9
+    Time.zone.today.year - first_school_year +
+      (Time.zone.today.month >= 9 ? 1 : 0)
+  end
```

Comment:
I get the feeling this can be golfed but 🚢

Reply:
immediately and then applies the two conditional increments:

```suggestion
  def year_group
    result = Time.zone.today.year - date_of_birth.year - 5
    result -= 1 if date_of_birth.month >= 9
    result += 1 if Time.zone.today.month >= 9
    result
  end
```

Accept it if you think it's clearer, reject if not 👍

Reply:
Ya, 🚢 for now I say. Once we have the baseline set we can start playing golf with it. 😄